### DISCUSSION TITLE
Will refetchInterval trigger new queries if a query takes a long time?

### DISCUSSION THREAD

**Discussion by dmitriid** [TanStack/query#2600] [Q&A] [answered]
(an aside: it's probably easy to just run and check in actual code than to ask 😄  )

If a query takes a longer time to fetch data than the `refetchInterval`, will React Query trigger an additional refetch, or will it wait for the query to complete?

E.g.:

```js
useQuery(this_fetch_takes_10_seconds_sometimes, {
   refetchInterval: 5 * 1000 /* 5 seconds */
});

**Comment by karol-janik** [comment, marked as answer]
No 😄  Will w8 till initial fetching finish.

**Reply by dmitriid** [reply]
Thank you! 

(Funnily enough I never got around to writing the relevant code yesterday :) )
